Add fetch rebase docs

dhruvbalwada opened the pull request:
Added some info about fetching and rebasing for after the PR is accepted.

jbusecke commented on docs/contributing.rst:
```diff
@@ -148,3 +148,12 @@ Preparing Pull Requests
    The merged pull request will undergo the same testing that your local branch
    had to pass when pushing.
+
+#. After your pull request is merged into the `GliderTools/master`, you will need
```
I think it might be helpful to explicitly say that this should also be done right **before** someone is planning changes.

jbusecke commented on docs/contributing.rst:
```diff
@@ -148,3 +148,12 @@ Preparing Pull Requests
+   version of GliderTools.
+
+    $ git checkout master
+    $ git fetch upstream
```
Does the guide include instructions on how to set up the `upstream` remote?

jbusecke replied:
Also could you add some explanations what is happening in each step:
git checkout master #switching back to the main/master branch
git fetch upstream #Downloading all the changes from the central repo
git rebase upstream/master #applying the changes that were made to the original repo's master branch onto your local branch, so they match exactly.

dhruvbalwada replied:
Yes, setting up of upstream master are explained in step 2.

jbusecke commented on glidertools/_version.py:
```diff
@@ -0,0 +1 @@
+__version__ = "2021.4.dev25+g01092e3"
```
This file should not be committed. I will add it to the .gitignore, but can you remove it here?

jbusecke replied:
Actually I am not sure If I should add it there...but can you still remove it?

jbusecke commented:
I believe as soon as my small nitpicks are addressed this is ready to merge. Thanks for the work @dhruvbalwada

dhruvbalwada commented:
I have addressed your comments. Hopefully this is good to go. 

I removed the _version file. There are also other files in there that start with _ , should be taken care of in gitignore if they are not needed.

dhruvbalwada commented:
I don't know why one of the pre-commit fails.

jbusecke reviewed (approved):
Thanks a lot. Looks good to me.
